Add [10.1021/jacsau.4c00276]  polymers to the polyply library

ricalessandri opened the pull request:
Contributing the 3x phthalimide-containing polymer models (PMAP, PEPP, PVBP) developed in [this recent paper](https://doi.org/10.1021/jacsau.4c00276) to the polyply library.

As part of this, I'm also starting to use common termini, e.g., the same CH3 termini. 

Things left:
- [x] add new polymers to the [library md doc](https://github.com/marrink-lab/polyply_1.0/blob/master/LIBRARY.md)
- [x] remove the [previous CH3.ff](https://github.com/marrink-lab/polyply_1.0/blob/master/polyply/data/oplsaaLigParGen/CH3.ff) and update the [PTMA model](https://github.com/marrink-lab/polyply_1.0/blob/master/polyply/data/oplsaaLigParGen/PTMA.oplsaa.LigParGen.ff) so that it uses the new one; probably, we'll have then to update the PTMA test
- [x] use the CH3.ff also for the new polymers + add tests for the new polymers

fgrunewald reviewed (changes requested):
Ideally we would only have one CH3ter and change the charges or atomtype once it get's connected via a different residue / bond. Do you think that would be possible?

ricalessandri commented:
Right, that makes sense. Does https://github.com/marrink-lab/polyply_1.0/pull/327 handle this automagically? My itp_to_ff version does not. I could do it by hand for these 3 polymers.

fgrunewald commented:
@ricalessandri yes, but there are some edge cases I haven't considered yet. For example, when CH3 is bound to acrylate either via the backbone or via the oxygen.

fgrunewald commented:
@ricalessandri is this ready to be merged?

ricalessandri commented:
In [25f992e](https://github.com/marrink-lab/polyply_1.0/pull/378/commits/25f992e1c101c1d2b29c468694c00c60f67436b6), I'm trying the approach of a unique `CH3.ff`. First off, I tried to to this for PTMA (for that, I actually currently define `CH3a` and `CH3b`, whose difference is just charges). If this works, then I can do the same for the new redox polymers and then this PR is done. However, the test is failing? Can you take a quick look to check what I'm doing wrong, @fgrunewald ?

fgrunewald commented on polyply/data/oplsaaLigParGen/links_PTMA.ff:
```diff
@@ -119,9 +123,13 @@ resname "CH3a|PTMA"
 [ atoms ]
-  C17  {"replace": {"charge": 0.1111}} 
+  C17  {"replace": {"charge":  0.1111}} 
+  -C1  {"replace": {"charge": -0.1854}}
+  -H2  {"replace": {"charge":  0.0386}}
+  -H3  {"replace": {"charge":  0.0386}}
+  -H4  {"replace": {"charge":  0.0386}}
```
@ricalessandri I think you would want to have these as +C1 and +H2 etc. Now they are referring to the previous residue, which I think is incorrect.

ricalessandri replied:
yes, indeed

ricalessandri commented:
Nice, test for PTMA passed. What I had to do:

1. Fix what @fgrunewald suggested.
2. Fix some other problem I found. Namely, the previous version of the CH3b link .ff file actually contained a mistake that, _for some reason I don't fully understand_, was not showing up in the output itp file. But now that I made a single CH3.ff, it was showing up, namely, there were 2 times the internal angles of the CH3 block "on the right". So 3 extra lines in the final .itp file that were making the test still fail after I fixed point 1.

Final note: I've fixed the LIBRARY.md conflict following the format I suggest in https://github.com/marrink-lab/polyply_1.0/pull/411.

ricalessandri commented:
PR should be ready to be merged now @fgrunewald !
